### PR TITLE
#101 - Let simulate return a struct

### LINKED ISSUE #101: Let simulate return a struct

### PR DESCRIPTION
**Pull request by schillic**
Closes #101.

I do not particularly like the name `inputs` because we have control inputs and nondeterministic inputs. Is there a better word?

### CODE REVIEW

**Review by mforets — APPROVED** [review]

**Comment by mforets** [comment]
>  because we have control inputs and nondeterministic inputs.

i think "controlled" (or control) inputs vs "uncontrolled " / external inputs is fairly standard

**Comment by schillic** [comment]
Thanks, so what is the proposal? `controlled_inputs` and `uncontrolled_inputs`? `controls` and `external_inputs`?

File: src/init_DifferentialEquations.jl [hunk 1]
```diff
@@ -1,6 +1,41 @@
...
+    solutions = @inbounds [SimulationSolution(simulations[i], controls[i],
+        isassigned(inputs, i) ? inputs[i] : nothing)
+        for i in eachindex(simulations)]
+    return EnsembleSimulationSolution(solutions)
+end
+
+Base.length(ess::EnsembleSimulationSolution) = length(ess.solutions)
+solution(ess::EnsembleSimulationSolution, i) = ess.solutions[i]
+trajectory(ess::EnsembleSimulationSolution, i) = trajectory(solution(ess, i))
+trajectories(ess::EnsembleSimulationSolution) = trajectory.(ess.solutions)
```
Comment (schillic) [inline src/init_DifferentialEquations.jl:35]: I am not sure about this name. `trajectories` returns the solutions, not just the trajectories. Should it rather be called `solutions`?

**Comment by mforets** [comment]
i would keep it as is. i think that the getter functions

```julia
controls(sol::SimulationSolution) = sol.controls
inputs(sol::SimulationSolution) = sol.inputs
```
are fine. but we shall now or later add docstrings to those methods to clear possible doubts.

File: src/init_DifferentialEquations.jl (hunk 1 above)
Comment (mforets) [reply src/init_DifferentialEquations.jl:35]: does it? i expect `trajectory.(ess.solutions)`  to return a vector of all computed trajectories

Comment (schillic) [reply src/init_DifferentialEquations.jl:35]: Oh, you are right :facepalm:

**Pull request merged**